wfe/web-front-end.go

```diff
@@ -207,13 +211,20 @@ func (wfe *WebFrontEndImpl) NewCertificate(response http.ResponseWriter, request
 
 	var init core.CertificateRequest
 	if err = json.Unmarshal(body, &init); err != nil {
+		fmt.Println(err)
 		sendError(response, "Error unmarshaling certificate request", http.StatusBadRequest)
 		return
 	}
 
-	wfe.log.Notice(fmt.Sprintf("Asked to create new certificate: %v %v", init, key))
+	wfe.log.Notice(fmt.Sprintf("Client requested new certificate: %v %v %v",
+		request.RemoteAddr, init, key))
 
 	// Create new certificate and return
+	// TODO IMPORTANT: The RA trusts the WFE to provide the correct key. If the
+	// WFE is compromised, *and* the attacker knows the public key of an account
```

Comment:
But if the WFE is compromised, game's over anyway, one way or another. I think it's OK to open an issue to do a secondary validation for paranoia, but this sort of thing is what the Activity Monitor is for. That's where the double-check should occur, so that it can perform an emergency stop.